Rally proccess(es) keep running after aborting on error

**Rally version** (get with `esrally --version`): 
rally 1.0.1.dev0 (git revision: 7700a36)

**Invoked command**:
`./rally --distribution-version=6.4.0  --track-path=<non-existing-path> --report-file=<path> --preserve-install=true --on-error=abort`

**Configuration file (located in `~/.rally/rally.ini`))**:
```
[meta]
config.version = 17

[system]
env.name = local

[node]
root.dir = /Users/simitt/.rally/benchmarks
src.root.dir = /Users/simitt/.rally/benchmarks/src

[source]
remote.repo.url = https://github.com/elastic/elasticsearch.git
elasticsearch.src.subdir = elasticsearch

[benchmarks]
local.dataset.cache = /Users/simitt/.rally/benchmarks/data

[reporting]
datastore.type = in-memory
datastore.host =
datastore.port =
datastore.secure =
datastore.user =
datastore.password =

[tracks]
default.url = https://github.com/elastic/rally-tracks

[teams]
default.url = https://github.com/elastic/rally-teams

[defaults]
preserve_benchmark_candidate = False

[distributions]
release.cache = true
```

**JVM version**:
java 10.0.2 2018-07-17

**OS version**:
macOs 10.13.6

**Description of the problem including expected versus actual behavior**:
Rally ends with an exception as the `track-path` does not exist. When waiting a bit and restarting rally again, it returns following error:
```
[ERROR] Cannot race. There are other Rally processes running on this machine (PIDs: [26755]) but only one Rally benchmark is allowed to run at the same time. Please check and terminate these processes and retry again.
```
Checking which process it is that is still running shows:
```
rally/.venv/bin/python3 rally/.venv/bin/esrally --distribution-version=6.4.0 --track-path=<non-existing-path> --report-file=<path> --preserve-install=true --on-error=abort
```

**Steps to reproduce**:
 1. start rally with an invalid `track-path`
 2. wait until it fails, and give it some time to shut down
 3. start rally again

--- a/esrally/racecontrol.py
+++ b/esrally/racecontrol.py
@@ -178,8 +178,6 @@ class BenchmarkActor(actor.RallyActor):
 
     def setup(self, msg, sender):
         self.start_sender = sender
-        self.mechanic = self.createActor(mechanic.MechanicActor, targetActorRequirements={"coordinator": True})
-
         self.cfg = msg.cfg
         # to load the track we need to know the correct cluster distribution version. Usually, this value should be set but there are rare
         # cases (external pipeline and user did not specify the distribution version) where we need to derive it ourselves. For source
@@ -211,6 +209,7 @@ class BenchmarkActor(actor.RallyActor):
         self.race_store = metrics.race_store(self.cfg)
         self.logger.info("Asking mechanic to start the engine.")
         cluster_settings = challenge.cluster_settings
+        self.mechanic = self.createActor(mechanic.MechanicActor, targetActorRequirements={"coordinator": True})
         self.send(self.mechanic, mechanic.StartEngine(self.cfg, self.metrics_store.open_context, cluster_settings, msg.sources, msg.build,
                                                       msg.distribution, msg.external, msg.docker))
 

